### PR TITLE
fix: adding branch name in the export application api

### LINKED ISSUE #24486: [Bug]: The export App does not have the latest data when the default branch is updated
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Steps To Reproduce

- Connect an app to git
- Change the default branch
- Refresh the branch and delete the old default branch (both git and locally)
- Now make changes in the new default branch app
- Export the App and notice latest changes are missing

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Latest

### PR DESCRIPTION
**Pull request by brayn003**
## Description
Fixes the export application api by adding the branch name

#### PR fixes following issue(s)
Fixes #24486 

#### Type of change
- Bug fix (non-breaking change which fixes an issue)
>
>
- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by brayn003** [comment]
Could not write Cypress tests because there is no Rest API to change default branch on Gitea

**Comment by AnaghHegde** [comment]
> Could not write Cypress tests because there is no Rest API to change default branch on Gitea

You can create a feature branch and make changes in the feature branch and then call the export Application API. This is enough to verify that export app flow supports branch name parameter.

**Comment by Parthvi12** [comment]
@brayn003 Exporting application is failing with this error, cypress tests are also failing because of same reason can you please check: 
Screenshot 2023-08-10 at 10 16 14 AM

**Comment by brayn003** [comment]
Checking

**Comment by Parthvi12** [comment]
@brayn003 I tried the steps mentioned in the issue and when I export the app and re-import it, I am not seeing latest changes I did after changing default branch & deleting previous default branch, adding video:

https://github.com/appsmithorg/appsmith/assets/80334441/bd607bc6-89a0-4f08-9161-c2c93a3d23eb

**Comment by Parthvi12** [comment]
@brayn003 Verified the fix, its working fine.

**Comment by Parthvi12** [comment]
> > Could not write Cypress tests because there is no Rest API to change default branch on Gitea
> 
> You can create a feature branch and make changes in the feature branch and then call the export Application API. This is enough to verify that export app flow supports branch name parameter.

@brayn003 Are we planning to add test?

**Comment by nayan-rafiq** [comment]
@brayn003 In other APIs, we send the branch name as a header param. I can see that here you're sending it as a request parameter and also changing the server code. Is there any complexity to get the branch name in client (it should be somewhere there as other API calls need it) and send it to API as a header param?

**Comment by brayn003** [comment]
@nayan-rafiq We cannot use the header as the API is opened in the browser as a url

File: app/client/src/pages/Applications/ApplicationCard.tsx [hunk 1]
```diff
@@ -473,7 +473,8 @@ export function ApplicationCard(props: ApplicationCardProps) {
     existingLink && existingLink.remove();
     const link = document.createElement("a");
 
-    link.href = getExportAppAPIRoute(applicationId);
```
Comment (nayan-rafiq) [inline app/client/src/pages/Applications/ApplicationCard.tsx:476]: @brayn003 Is there any other way to grab the branch name except parsing it from URL? Is it possible to get it from the headers as other APIs?

**Comment by brayn003** [comment]
@Parthvi12 added tests

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug24486_Spec.ts
Comment (AnaghHegde) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug24486_Spec.ts:1]: Why does the cypress test name is Bug24486? Why can't we keep this as a normal test scenario under the export App flow?

Comment (Parthvi12) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug24486_Spec.ts:1]: @AnaghHegde We add tests related to bugs separately in bugTests to keep track of what all bugs have been automated

Comment (brayn003) [reply app/client/cypress/e2e/Regression/ClientSide/BugTests/Bug24486_Spec.ts:1]: This test only covers the scenarios in the issue posted. Will remove this in the future to contain more scenarios, then we can use this as a standard flow test

**Pull request merged**